tasks/linked-list/src/main/java/com/CustomLinkedList.java

```diff
@@ -0,0 +1,269 @@
+     */
+    private Node<E> node(int index) {
+        Node<E> node;
+        if (size / 2 > index) {
+            node = this.first;
+
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+        } else {
+            node = this.last;
+
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+        }
+
+        return node;
+    }
+
+    /**
+     * Returns the first element in this list.
+     *
+     * @return the first element in this list
+     * @throws NoSuchElementException if this list is empty
+     */
+    public E getFirst() {
+        Node<E> first = this.first;
+        if (first == null)
+            throw new NoSuchElementException();
+
+        return first.item;
+    }
+
+    /**
+     * Returns the last element in this list.
+     *
+     * @return the last element in this list
+     * @throws NoSuchElementException if this list is empty
+     */
+    public E getLast() {
+        Node<E> last = this.last;
+        if (last == null)
+            throw new NoSuchElementException();
+
+        return last.item;
+    }
+
+    /**
+     * Removes and returns the first element from this list.
+     *
+     * @return the first element from this list
+     * @throws NoSuchElementException if this list is empty
+     */
+    public E removeFirst() {
+        Node<E> first = this.first;
+
+        if (first == null)
+            throw new NoSuchElementException();
+
+        Node<E> next = first.next;
+        this.first = next;
+        if (next != null) {
+            next.prev = null;
+        } else {
+            this.last = null;
+        }
+
+        size--;
+        return first.item;
+    }
+
+    /**
+     * Removes and returns the last element from this list.
+     *
+     * @return the last element from this list
+     * @throws NoSuchElementException if this list is empty
+     */
+    public E removeLast() {
+        Node<E> last = this.last;
+
+        if (last == null)
+            throw new NoSuchElementException();
+
+        Node<E> prev = last.prev;
+        this.last = prev;
+        if (prev != null) {
+            prev.next = null;
+        } else {
+            this.first = null;
+        }
+
+        size--;
+        return last.item;
+    }
+
+    /**
+     * Removes the element at the specified position in this list.
+     * Shifts any subsequent elements to the left (subtracts one from their indices).
+     *
+     * @param index the index of the element to be removed
+     * @return the element that was removed from the list
+     * @throws IndexOutOfBoundsException if the index is out of range ({@code index < 0 || index >= size()})
+     */
+    public E remove(int index) {
+        if (index < 0 || index >= size)
+            throw new IndexOutOfBoundsException("Index %d, Size: %d".formatted(index, size));
+
+        if (index == 0) {
+            return removeFirst();
+        } else if (index == size - 1) {
```

Comment:
The method doesn't decrement the size counter when removing elements from the middle of the list (only when removing first or last elements). This will cause the size to be incorrect.